flutterproject/flutterproject.go

```diff
@@ -0,0 +1,220 @@
+		if entries, err := p.fileManager.ReadDirEntryNames(testsDirPath); err == nil && len(entries) > 0 {
+			for _, entry := range entries {
+				if strings.HasSuffix(entry, "_test.dart") {
+					hasTests = true
+					break
+				}
+			}
+		}
+	}
+
+	if !hasTests {
+		testsDirPath = ""
+	}
+
+	return testsDirPath
+}
+
+func (p *Project) IOSProjectPth() string {
+	const iosProjectRelPth = "ios/Runner.xcworkspace"
+
+	hasIOSProject := false
+	iosProjectPth := filepath.Join(p.rootDir, iosProjectRelPth)
+	if exists, err := p.pathChecker.IsPathExists(iosProjectPth); err == nil && exists {
+		hasIOSProject = true
+	}
+
+	if !hasIOSProject {
+		iosProjectPth = ""
+	}
+
+	return iosProjectPth
+
+}
+
+func (p *Project) AndroidProjectPth() string {
+	const androidProjectRelPth = "android/build.gradle"
+	const androidProjectKtsRelPth = "android/build.gradle.kts"
+
+	hasAndroidProject := false
+	androidProjectPth := filepath.Join(p.rootDir, androidProjectRelPth)
+	if exists, err := p.pathChecker.IsPathExists(androidProjectPth); err == nil && exists {
+		hasAndroidProject = true
+	}
+
+	if !hasAndroidProject {
+		androidProjectPth = filepath.Join(p.rootDir, androidProjectKtsRelPth)
+		if exists, err := p.pathChecker.IsPathExists(androidProjectPth); err == nil && exists {
+			hasAndroidProject = true
+		}
+	}
+
+	if !hasAndroidProject {
+		androidProjectPth = ""
+	}
+
+	return androidProjectPth
+}
+
+func (p *Project) FlutterAndDartSDKVersions() (FlutterAndDartSDKVersions, error) {
+	sdkVersions := FlutterAndDartSDKVersions{}
+
+	fvmFlutterVersion, err := sdk.NewFVMVersionReader(p.fileManager).ReadSDKVersion(p.rootDir)
+	if err != nil {
+		return FlutterAndDartSDKVersions{}, err
+	} else {
+		sdkVersions.FVMFlutterVersion = fvmFlutterVersion
+	}
+
+	asdfFlutterVersion, err := sdk.NewASDFVersionReader(p.fileManager).ReadSDKVersions(p.rootDir)
+	if err != nil {
+		return FlutterAndDartSDKVersions{}, err
+	} else {
+		sdkVersions.ASDFFlutterVersion = asdfFlutterVersion
+	}
+
+	pubspecLockFlutterVersion, pubspecLockDartVersion, err := sdk.NewPubspecLockVersionReader(p.fileManager).ReadSDKVersions(p.rootDir)
+	if err != nil {
+		return FlutterAndDartSDKVersions{}, err
+	} else {
+		sdkVersions.PubspecLockFlutterVersion = pubspecLockFlutterVersion
+		sdkVersions.PubspecLockDartVersion = pubspecLockDartVersion
+	}
+
+	pubspecFlutterVersion, pubspecDartVersion, err := sdk.NewPubspecVersionReader(p.fileManager).ReadSDKVersions(p.rootDir)
+	if err != nil {
+		return FlutterAndDartSDKVersions{}, err
+	} else {
+		sdkVersions.PubspecFlutterVersion = pubspecFlutterVersion
+		sdkVersions.PubspecDartVersion = pubspecDartVersion
+	}
+
+	return sdkVersions, nil
+}
+
+func (p *Project) FlutterSDKVersionToUse() (string, error) {
+	sdkVersions, err := p.FlutterAndDartSDKVersions()
+	if err != nil {
+		return "", err
+	}
+
+	sdkQuery := createSDKQuery(sdkVersions)
+	release, err := fluttersdk.FindLatestRelease(fluttersdk.MacOS, fluttersdk.ARM64, fluttersdk.Stable, sdkQuery)
```

Comment:
We might be having a problem if we hardcode `fluttersdk.ARM64` while we still have some Intel machine. For example this Flutter version:
```
{
       "hash": "77d935af4db863f6abd0b9c31c7e6df2a13de57b",
       "channel": "stable",
       "version": "2.8.1",
       "dart_sdk_version": "2.15.1",
       "dart_sdk_arch": "x64",
       "release_date": "2021-12-16T19:39:07.366650Z",
       "archive": "stable/macos/flutter_macos_2.8.1-stable.zip",
       "sha256": "90a9f4e05b9e7c22e6c646a2d275a5d6b48497cf7e24accbb62b0b9692687dbb"
     },
```
does not have an arm64 version and if the users flutter requirements is strict and can only use this given version then the step will not find it. 

I checked the newer Flutter versions and all of the SDK version are available on both architectures so there this is not a problem.